src/views/shared/snackbar/snackbar.view.tsx

```diff
@@ -4,16 +4,17 @@ import { ReactComponent as ErrorIcon } from "src/assets/icons/error.svg";
 import { ReactComponent as SuccessIcon } from "src/assets/icons/success.svg";
 import { ReactComponent as CloseIcon } from "src/assets/icons/xmark.svg";
 import { SNACKBAR_AUTO_HIDE_DURATION } from "src/constants";
-import { Message } from "src/domain";
+import { Env, Message } from "src/domain";
 import { useSnackbarStyles } from "src/views/shared/snackbar/snackbar.styles";
 
 interface SnackbarProps {
   message: Message;
   onClose: () => void;
-  onReport: (error: string) => void;
+  onReport: (error: string, reportForm: Extract<Env["reportForm"], { isEnabled: true }>) => void;
+  reportForm: Env["reportForm"];
```

Comment:
```diff
diff --git a/src/views/shared/snackbar/snackbar.view.tsx b/src/views/shared/snackbar/snackbar.view.tsx
index 4abd615..6560fc5 100644
--- a/src/views/shared/snackbar/snackbar.view.tsx
+++ b/src/views/shared/snackbar/snackbar.view.tsx
@@ -4,14 +4,14 @@ import { ReactComponent as ErrorIcon } from "src/assets/icons/error.svg";
 import { ReactComponent as SuccessIcon } from "src/assets/icons/success.svg";
 import { ReactComponent as CloseIcon } from "src/assets/icons/xmark.svg";
 import { SNACKBAR_AUTO_HIDE_DURATION } from "src/constants";
-import { Env, Message } from "src/domain";
+import { Message, ReportFormEnvEnabled } from "src/domain";
 import { useSnackbarStyles } from "src/views/shared/snackbar/snackbar.styles";
 
 interface SnackbarProps {
   message: Message;
   onClose: () => void;
-  onReport: (error: string, reportForm: Extract<Env["reportForm"], { isEnabled: true }>) => void;
-  reportForm: Env["reportForm"];
+  onReport: (error: string, reportForm: ReportFormEnvEnabled) => void;
+  reportForm: ReportFormEnvEnabled;
 }
 
 export const Snackbar: FC<SnackbarProps> = ({ message, onClose, onReport, reportForm }) => {
@@ -52,16 +52,14 @@ export const Snackbar: FC<SnackbarProps> = ({ message, onClose, onReport, report
         <div className={classes.wrapper}>
           <Icon message={message} />
           <p className={classes.message}>{text}</p>
-          {reportForm.isEnabled && (
-            <button
-              className={classes.reportButton}
-              onClick={() => {
-                onReport(parsed, reportForm);
-              }}
-            >
-              Report
-            </button>
-          )}
+          <button
+            className={classes.reportButton}
+            onClick={() => {
+              onReport(parsed, reportForm);
+            }}
+          >
+            Report
+          </button>
           <button className={classes.closeButton} onClick={onClose}>
             <CloseIcon className={classes.closeIcon} />
           </button>
```

Reply:
I prefer to pass the entire entity to the snackbar instead of just the `ReportFormEnvEnabled ` and do the check on top, but I've updated the `onReport` function signature.

Reply:
Right! What I proposed was incorrect, since it prevented showing the snackbar when `reportForm.isEnabled` was false, while we want to show it constantly, changing the msg and removing the button.